### PR TITLE
iOS: fix retain cycles in `EnvoyNetworkMonitor`

### PR DESCRIPTION
**Pull request by jpsim**
And properly clean up reachability ref.

Risk Level: Low
Testing: Validated using the memory graph debugger in Xcode with both reachability and path monitor modes
Docs Changes: N/A
Release Notes: N/A

Signed-off-by: JP Simard <jp@jpsim.com>

### CODE REVIEW

File: library/objective-c/EnvoyNetworkMonitor.m [hunk 1]
```diff
@@ -38,6 +43,7 @@ - (void)startPathMonitor {
   nw_path_monitor_set_queue(_path_monitor, queue);
 
   __block envoy_network_t previousNetworkType = -1;
+  envoy_engine_t engineHandle = _engineHandle;
```
Comment (Augustyniak) [inline library/objective-c/EnvoyNetworkMonitor.m:46]: Why was this needed?

**Review by Augustyniak — APPROVED** [review]

File: library/objective-c/EnvoyNetworkMonitor.m (hunk 1 above)
Comment (jpsim) [reply library/objective-c/EnvoyNetworkMonitor.m:46]: It was causing a retain cycle in the nw update block.
